scripts/extract-ios-device-names.ts

```diff
@@ -81,21 +81,12 @@ const formatOutput = async (unformatted: string) => {
 };
 
 export async function extractIOSDeviceNames() {
-  try {
-    if (fs.statSync(outputPath)) {
-      // Out with the old, in with the new
-      fs.unlinkSync(outputPath);
-    }
-  } catch (e) {
-    // File does not exists, carry along
-  }
-
   const files = await getDefinitionFiles();
   const definitions = await collectDefinitions(files);
   const formatted = await formatOutput(
     template(JSON.stringify(definitions, undefined, 2))
   );
 
+  // This will always overwrite the file with a fresh version
   fs.writeFileSync(outputPath, formatted);
-  console.log('✅ Regenerated ios-device-list.tsx');
```

Comment:
this still has a potential race as well -- if the process is interrupted mid-write you'll get a partially written file

usually the fix for this is to:
1. write to a temporary file next to it
2. `mv` the temporary file overtop the old file (this action being atomic)

though I'm not sure if that matters for this

https://youtu.be/-9eXCb3yvyY

Reply:
@asottile-sentry Yes correct. Doesnt the`write to a temporary file next to it` step expose us to having that file stuck around not paying rent in case the process is interrupted between steps 1 and 2?

>though I'm not sure if that matters for this
I dont think it does, this file is gitignored by default, we just want to make sure it doesnt get removed

Love the yt video :D

Reply:
@JonasBa yeah it does potentially expose us to the file sticking around, though typically you'd have a `finally { ... }` to clean up the file in the interruption case (or in python a context manager such as `with tempfile.NamedTemporaryFile(dir=...)`

Reply:
Ah yeah, it's a bit more annoying with node. I just pushed an improvement as per your suggestion :) thanks for the suggestion